turtlesim/CMakeLists.txt

```diff
@@ -13,35 +13,54 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(ament_index_cpp REQUIRED)
 find_package(geometry_msgs REQUIRED)
-find_package(Qt5 REQUIRED COMPONENTS Widgets)
+
+find_package(Qt6 COMPONENTS Widgets QUIET)
+set(QT_VERSION_MAJOR 6)
+if(NOT Qt6_FOUND)
+  find_package(Qt5 REQUIRED COMPONENTS Widgets)
+  set(QT_VERSION_MAJOR 5)
+endif()
```

Comment:
I'm somewhat concerned about this logic.

In particular, I think we want to continue to use Qt5 right now, but have everything in place so that we could switch to Qt6 in the future.

Because of the way we build binary packages, we will only install Qt5 libraries, and thus we'll continue to build against Qt5 for binaries.

But in from-source builds, it depends on what the user has installed.  Thus, I'm a bit afraid that we'll get bug reports from users saying things won't compile properly due to Qt6 being installed.  And I don't think we are quite ready for that.

So my suggestion here is to hide this whole thing behind a CMake option, something like `USE_QT6`.  The the logic would be something like:

```
if (USE_QT6)
  find_package(Qt6 REQUIRED COMPONENTS Widgets)
  set(QT_VERSION_MAJOR 6)
else()
  find_package(Qt5 REQUIRED COMPONENTS Widgets)
  set(QT_VERSION_MAJOR 5)
endif()
```

What do you think?